PokeAlarm/Manager.py

```diff
@@ -893,8 +893,10 @@ def process_egg(self, egg):
         team_id = self.__cache.get_gym_team(gym_id)
         gym_info = self.__cache.get_gym_info(gym_id)
 
+        if egg['gym_name'] == 'unknown':  # if RM or empty
+            egg.update({"gym_name": gym_info['name']})
```

Comment:
Update shouldn't be used for a single key value pair. The correct way would be `egg['name'] = gym_info['name']`.